js/limit-subtheme-connections.js

```diff
@@ -1,83 +1,61 @@
-    
-    // create a binding for when the metabox changes
-    watchMetaboxOnChange(subCategoryCategoryMetabox, subCategoryCategoryMetaboxWrapper, subCategoryCategoryConnectionTableRow, postPublishButton, $);
+document.addEventListener('DOMContentLoaded', () => {
+    const selectors = {
+        subCategoryCategoryMetaboxWrapper: '#p2p-to-pdc-category_to_pdc-subcategory',
+        subCategoryCategoryMetabox: 'div[data-p2p_type="pdc-category_to_pdc-subcategory"]',
+        subCategoryCategoryConnectionTableRow: 'div[data-p2p_type="pdc-category_to_pdc-subcategory"] > table.p2p-connections > tbody > tr',
+        postPublishButton: 'div.edit-post-header__settings > button.editor-post-publish-button__button',
+    };
 
-    // wait before the dom is loaded
-    metaboxValidationAfterPageLoad(
-        subCategoryCategoryMetabox, 
-        subCategoryCategoryMetaboxWrapper, 
-        subCategoryCategoryConnectionTableRow, 
-        postPublishButton, 
-        $
-    );
+    const {
+        subCategoryCategoryMetabox,
+        subCategoryCategoryMetaboxWrapper,
+        subCategoryCategoryConnectionTableRow,
+        postPublishButton,
+    } = selectors;
+
+    observeMetaboxChanges(subCategoryCategoryMetabox, subCategoryCategoryMetaboxWrapper, subCategoryCategoryConnectionTableRow, postPublishButton);
+    validateConnectionsOnLoad(subCategoryCategoryMetabox, subCategoryCategoryMetaboxWrapper, subCategoryCategoryConnectionTableRow, postPublishButton);
 });
 
-/**
- * Validate the PostsToPosts connections metaboxes after pageload.
- * 
- * @param {string} subCategoryCategoryMetabox 
- * @param {string} subCategoryCategoryMetaboxWrapper 
- * @param {string} subCategoryCategoryConnectionTableRow 
- * @param {string} postPublishButton 
- * @param {Object} $ 
- */
-function metaboxValidationAfterPageLoad(
-    subCategoryCategoryMetabox, 
-    subCategoryCategoryMetaboxWrapper, 
-    subCategoryCategoryConnectionTableRow, 
-    postPublishButton, 
-    $
-)
-{
-    setTimeout(function(){ 
-        if($(subCategoryCategoryConnectionTableRow).length = 1)
-        {
-            $(postPublishButton).prop("disabled",false);
-            $(subCategoryCategoryMetabox + '> div.p2p-create-connections').hide();
-            $(subCategoryCategoryMetaboxWrapper).css("border", "");
-        }
+function observeMetaboxChanges(metabox, metaboxWrapper, connectionRow, postPublishButton) {
+    const observer = new MutationObserver(() => {
+        validateConnections({ metabox, metaboxWrapper, connectionRow, postPublishButton });
+    });
 
-        if($(subCategoryCategoryConnectionTableRow).length == 0)
-        {
-            $(postPublishButton).prop("disabled",true);
-            $(subCategoryCategoryMetabox + '> div.p2p-create-connections').show();
-            $(subCategoryCategoryMetaboxWrapper).css("border", "1px solid red");
-        }
-    }, 1000);
+    const config = { childList: true, subtree: true };
```

Comment:
Van de drie files, is dit de enige die de config in een losse variable heeft staan. 
laten we dit consistent houden over die drie files. (mierenneuken)